core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.kt

```diff
@@ -0,0 +1,159 @@
+import com.jakewharton.threetenabp.AndroidThreeTen
+import org.kiwix.kiwixmobile.core.data.local.KiwixDatabase
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.di.components.DaggerCoreComponent
+import org.kiwix.kiwixmobile.core.downloader.DownloadMonitor
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+
+abstract class CoreApp : Application() {
+  companion object {
+    @JvmStatic
+    lateinit var instance: CoreApp
+      private set
+
+    @JvmStatic
+    lateinit var coreComponent: CoreComponent
+
+    init {
+      AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
+    }
+  }
+
+  @Inject
+  lateinit var downloadMonitor: DownloadMonitor
+
+  @Inject
+  lateinit var nightModeConfig: NightModeConfig
+
+  @Inject
+  lateinit var kiwixDatabase: KiwixDatabase
+
+  /**
+   * The init of this class does the work of initializing,
+   * simply injecting it is all that there is to be done
+   */
+  @Inject
+  internal lateinit var jniInitialiser: JNIInitialiser
+
+  override fun attachBaseContext(base: Context) {
+    super.attachBaseContext(base)
+    if (BuildConfig.DEBUG) {
+      MultiDex.install(this)
+    }
+  }
+
+  override fun onCreate() {
+    super.onCreate()
+    instance = this
+    coreComponent = DaggerCoreComponent.builder()
+      .context(this)
+      .build()
+    AndroidThreeTen.init(this)
+    writeLogFile()
+    coreComponent.inject(this)
+    kiwixDatabase.forceMigration()
+    downloadMonitor.init()
+    nightModeConfig.init()
+    configureStrictMode()
+  }
+
+  private fun configureStrictMode() {
+    if (BuildConfig.DEBUG) {
+      StrictMode.setThreadPolicy(
+        StrictMode.ThreadPolicy.Builder().apply {
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            detectResourceMismatches()
+          }
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            detectUnbufferedIo()
+          }
+          detectCustomSlowCalls()
+          detectDiskReads()
+          detectDiskWrites()
+          detectNetwork()
+          penaltyFlashScreen()
+          penaltyLog()
+        }.build()
+      )
+      StrictMode.setVmPolicy(
+        VmPolicy.Builder().apply {
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            detectCleartextNetwork()
+          }
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            detectContentUriWithoutPermission()
+          }
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            detectFileUriExposure()
+          }
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            detectNonSdkApiUsage()
+          }
+          detectActivityLeaks()
+          detectLeakedClosableObjects()
+          detectLeakedSqlLiteObjects()
+          penaltyLog()
+          detectLeakedRegistrationObjects()
+        }.build()
+      )
+    }
+  }
+
+  private fun writeLogFile() {
```

Comment:
Can it be moved to some util?

Reply:
I think probably more suited to another ticket and maybe a rewrite? Honestly I don't think this feature has once been useful to me

Reply:
ticket [here](https://github.com/kiwix/kiwix-android/issues/2143)